Dockerfile

```diff
@@ -1,23 +1,29 @@
-FROM python:3.8-alpine as base
-RUN apk add --no-cache curl tini && \
-    adduser -u 1000 -g 1000 -D kyaku
-WORKDIR /home/kyaku/renku-notebooks
-
-FROM base as builder
-ENV POETRY_HOME=/opt/poetry
+FROM python:3.11-bullseye as builder
+RUN groupadd --gid 1000 renku && \
+    useradd --gid 1000 --uid 1000 --groups 100 --create-home renku && \
+    mkdir -p /app && \
+    chown -R 1000:1000 /app 
+USER 1000:1000
+WORKDIR /app
+RUN python3 -m pip install --user pipx && \
+    python3 -m pipx ensurepath && \
+    /home/renku/.local/bin/pipx install poetry && \
```

Comment:
@Panaetius I use pipx because I want poetry to be in a separate environment and I do not want to have poetry at all in the final image.

So either I use pipx or I create a separate virtualenv for poetry and then install poetry there. Pipx was easier to use. So I will leave this as is.